iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -10,26 +10,39 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport
 {
+    internal enum ClientTransportStatus
+    {
+        Closed,
```

Comment:
Let's add comments for each state, what it actually means in this class, and what behavior is expected in each state.

Reply:
Consider moving this and the code that changes states to its own class (state management class) so it can be unit tested.

Reply:
Added the comments.

Reply:
We will address the separation of state management in the subsequent "separate-delegating-handlers" PR.